transformations/transformer_text_generation/transformation.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import math
+
+import numpy as np
+import torch
+from transformers import GPT2LMHeadModel, pipeline
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
+    datefmt="%m/%d/%Y %H:%M:%S",
+    level=logging.INFO,
+)
+logger = logging.getLogger(__name__)
+
+
+def set_seed(seed: int, no_cuda: bool):
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    if not no_cuda:
+        torch.cuda.manual_seed_all(seed)
+
+
+class TransformerTextGeneration(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+
+    def __init__(
+        self,
+        eos: str = "</s>",
+        no_cuda: bool = False,
+        dataset="sst2",
+        labeled=True,
+        seed=42,
+    ):
+
+        set_seed(seed, no_cuda)
+
+        if labeled:
+            if dataset == "imdb":
+                model_text_generation: str = "jmamou/gpt2-medium-IMDB"
+                # model_sentiment_classification = 'aychang/roberta-base-imdb'
+                model_sentiment_classification = "textattack/roberta-base-imdb"
+            elif dataset == "sst2":
+                model_text_generation: str = "jmamou/gpt2-medium-SST-2"
+                model_sentiment_classification = (
+                    "textattack/roberta-base-SST-2"
+                )
+        else:
+            model_text_generation: str = "gpt2-xl"
+            model_sentiment_classification = None
+
+        self.eos = eos
+        device = -1 if no_cuda else 0
+
+        # initialize text generation pipeline
+        model = GPT2LMHeadModel.from_pretrained(model_text_generation)
```

Comment:
'model_text_generation' might be referenced before assignment as it's getting initialized in the 'if' block.